src/rewriter.coffee

```diff
@@ -267,7 +267,8 @@ exports.Rewriter = class Rewriter
       # Added support for spread dots on the left side: f ...a
       if (tag in IMPLICIT_FUNC and token.spaced or
           tag is '?' and i > 0 and not tokens[i - 1].spaced) and
-         (nextTag in IMPLICIT_CALL or nextTag is '...' or
+         (nextTag in IMPLICIT_CALL or 
+         (nextTag is '...' and @tag(i + 2) in IMPLICIT_CALL and not @findTagsBackwards(i, ['INDEX_START', '['])) or
```

Comment:
I guess with the current test layout you'd need to put some in:

- `arrays` (array spreads)
- `assignment` (array/object rest destructuring)
- `functions` (argument rest arguments)
- `function_invocation` (spread arguments)
- `objects` (object spreads) - though there are no object spread tests there currently?
- `ranges` (disambiguating spreads from ranges)
- `slicing_and_splicing` (disambiguating spreads from slices and splices)

That *is* quite spread out, but probably the easiest thing to do for now. We may want a new `rests_and_spreads` test file where we can move all the rest/spread tests so all these cases are in one place.